src/modules/flow/http-client/http-client.c

```diff
@@ -602,30 +690,44 @@ boolean_post_process(struct sol_flow_node *node, void *data, uint16_t port, uint
 /*
  * --------------------------------- string node -----------------------------
  */
+
 static int
-string_process_token(struct sol_flow_node *node,
-    struct sol_json_token *key, struct sol_json_token *value)
+string_process_json(struct sol_flow_node *node, const struct sol_str_slice slice)
 {
+    bool found = false;
     char *result = NULL;
+    enum sol_json_loop_reason reason;
+    struct sol_json_scanner sub_scanner;
+    struct sol_json_token sub_key, sub_value, token;
 
-    result = strndup(value->start + 1, value->end - value->start - 2);
-    if (result) {
-        return sol_flow_send_string_take_packet(node,
-            SOL_FLOW_NODE_TYPE_HTTP_CLIENT_STRING__OUT__OUT, result);
+    sol_json_scanner_init(&sub_scanner, slice.data, slice.len);
+    found = false;
+
+    SOL_JSON_SCANNER_OBJECT_LOOP (&sub_scanner, &token, &sub_key, &sub_value, reason) {
+        if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "value")) {
+            if (sol_json_token_get_type(&sub_value) == SOL_JSON_TYPE_STRING)
+                result = sol_json_token_get_unescaped_string_copy(&sub_value);
+            else
+                result = strndup(sub_value.start,
+                    sub_value.end - sub_value.start);
+            SOL_NULL_CHECK(result, -ENOMEM);
+            found = true;
+            break;
+        }
     }
 
-    return -ENOMEM;
+    SOL_EXP_CHECK(!found, -ENOENT);
+
+    return sol_flow_send_string_take_packet(node,
+        SOL_FLOW_NODE_TYPE_HTTP_CLIENT_STRING__OUT__OUT, result);
 }
 
 static int
-string_process_data(struct sol_flow_node *node,
-    struct sol_http_response *response)
+string_process_data(struct sol_flow_node *node, struct sol_buffer *buf)
 {
     char *result;
 
-    SOL_HTTP_RESPONSE_CHECK_API(response, -EINVAL);
-
-    result = strndup(response->content.data, response->content.used);
+    result = strndup(buf->data, buf->used);
```

Comment:
maybe follow the optimization done for json and use `sol_buffer_steal_or_copy()`?